src/text_formatter/allowed_Expression_Detectors.ts

```diff
@@ -1,105 +1,39 @@
-/**
-* --------------------------------------
-* @author: Ronja K.
-* ---------------------------------------
-*/
-
-
 // TO DO: USE TOKENIZATION INSTEAD OF INPUT
 
-
 /**
-* ---------CurrentExpressionDetector------------
-* ---------every element have their own detect-function--------
-* @param
-* @return
-*/
-export function detectExpressionElements(input: string) {
-	var formattedInputExpression; // this one should get formatted
-
-	input = detectBracketLeft(input);
-	input = detectBracketRight(input);
-	input = detectEquivalence(input);
-	input = detectImplication(input);
-	input = detectNegation(input);
-	input = detectConjunction(input);
-	input = detectDisjunction(input);
-	return input;
-}
-
-/** 
-* detect and mark left brackets
-* ---------------------------------------
-* @param
-* @return
-*/
-function detectBracketLeft(input: string) {
-	input = input.replace(/(\[|\()/g, " bracketLeft ");
-	return input;
+* ---------CurrentExpressionFormatter------------
+* @param expression written by a user
+* @return a formatted expression as String
+*/
+export function formatExpressionElements(expression: string) {
+	const formattedInputExpression: string = expression;
+	return formattedInputExpression
+		.replace(/(\[|\()/g, " bracketLeft ")
+		.replace(/(\]|\))/g, " bracketRight ")
+		.replace(/(<-->|<==>|<=>|<->)/g, " equivalence ")
+		.replace(/(->|-->|=>|==>)/g, " implication ")
+		.replace(/(neq|not|nicht|¬|-)/ig, " negation ")
+		.replace(/(and|und|&|∧)/ig, " conjunction ")
+		.replace(/(or|oder|\||∨)/ig, " disjunction ")
+		.trim()								// delete white spaces at the beginn and the end of an expression
+		.replace(/\s{1,}/g, ","); 			// each element of an expression should seperated with a comma
```

Comment:
Instead of a string `"A,B,C..."` I'd rather return an array `["A","B","C"...]`:

- It's more convenient to work with ("how many tokens are there?", "get a subset of tokens", "Iterate over the tokens"...).
- It's more memory-efficient. Most string-implementations will require one consecutive piece of memory. Lists usually don't. Also, every operation on a string produces a copy of said string. 
- `string[]` or `Token[]` the more expressive return type (compared to `string` with some constraints on it's structure).